service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/CarePlanMapper.java

```diff
@@ -0,0 +1,117 @@
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01ClinicalDocument1;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Component2;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Component3;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Component5;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01Section;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01StructuredBody;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.hl7.fhir.dstu3.model.CarePlan.CarePlanIntent.PLAN;
+import static org.hl7.fhir.dstu3.model.CarePlan.CarePlanStatus.ACTIVE;
+import static org.hl7.fhir.dstu3.model.IdType.newRandomUuid;
+
+@Component
+@AllArgsConstructor
+public class CarePlanMapper {
+    private final String SNOMED = "2.16.840.1.113883.2.1.3.2.4.15";
+    private final String INFORMATION_ADVICE_GIVEN = "1052951000000105";
+
+    public List<CarePlan> mapCarePlan(POCDMT000002UK01ClinicalDocument1 clinicalDocument,
+                                      Encounter encounter) {
+
+        POCDMT000002UK01StructuredBody structuredBody = getStructuredBody(clinicalDocument);
+
+        return Arrays.stream(structuredBody.getComponentArray())
+                .map(POCDMT000002UK01Component3::getSection)
+                .map(this::findCarePlanSections)
+                .flatMap(List::stream)
+                .map(section -> createCarePlanFromSection(section, encounter))
+                .collect(Collectors.toUnmodifiableList());
+    }
+
+    public CarePlan createCarePlanFromSection(POCDMT000002UK01Section cpSection, Encounter encounter) {
+        CarePlan carePlan = new CarePlan();
+        carePlan.setIdElement(newRandomUuid());
+        carePlan
+                .setIntent(PLAN)
+                .setSubject(encounter.getSubject())
+                .setSubjectTarget(encounter.getSubjectTarget())
+                .setStatus(ACTIVE)
+                .setContextTarget(encounter)
+                .setContext(new Reference(encounter))
+                .setPeriod(encounter.getPeriod());
+
+        if (cpSection.isSetLanguageCode()) {
+            carePlan.setLanguage(NodeUtil.getNodeValueString(cpSection.getLanguageCode()));
+        }
+        if (cpSection.isSetTitle()) {
+            carePlan.setTitle(NodeUtil.getNodeValueString(cpSection.getTitle()));
+        }
+
+        if (cpSection.getText().sizeOfContentArray() > 0) {
+            String cpTextContent = NodeUtil.getNodeValueString(cpSection.getText().getContentArray(0));
+            Narrative narrative = new Narrative();
+            narrative.setDivAsString(cpTextContent);
+            narrative.setStatus(Narrative.NarrativeStatus.GENERATED);
+            if (cpSection.isSetText()) {
+                carePlan.setText(narrative);
+            }
+            carePlan.setDescription(cpTextContent);
+        }
+
+        // TODO 2020-07-02: CarePlan.Category required
```

Comment:
we can't mark a story as done and leave such TODO sections.
Please have a look at https://developer-test.nhs.uk/apis/cds-api-2-0-0/api_care_plan.html, maybe this helps a bit.
If you think that mapping for the given sections does not exist - that's fine, don't include these elements in the docs.

Reply:
These are non mandatory fields so I have removed the TODO